tests/test_mdspan_swap.cpp

```diff
@@ -0,0 +1,198 @@
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.rank_dynamic(), 2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.extent(0), 4);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.extent(1), 3);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.static_extent(0), dyn);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.static_extent(1), dyn);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.stride(0), 3);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.stride(1), 1);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.data_handle(), data1);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.rank(), 2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.rank_dynamic(), 2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.extent(0), 3);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.extent(1), 4);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.static_extent(0), dyn);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.static_extent(1), dyn);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.stride(0), 4);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.stride(1), 1);
+  });
+  ASSERT_EQ(errors[0], 0);
+  free_array(errors);
+}
+
+TEST(TestMDSpanSwap, std_swap_dynamic_extents) {
+  __MDSPAN_TESTS_RUN_TEST(test_mdspan_std_swap_dynamic_extents())
+}
+
+
+void test_mdspan_foo_swap_dynamic_extents() {
+  size_t* errors = allocate_array<size_t>(1);
+  errors[0] = 0;
+
+  dispatch([=] _MDSPAN_HOST_DEVICE () {
+    using map_t = Foo::layout_foo::template mapping<stdex::dextents<size_t ,2>>;
+    using acc_t = Foo::foo_accessor<int>;
+
+    int data1[12] = {1,2,3,4,5,6,7,8,9,10,11,12};
+    int data2[12] = {1,2,3,4,5,6,7,8,9,10,11,12};
+    int flag1 = 9;
+    int flag2 = 7;
+
+    stdex::dextents<size_t, 2> exts1(3,4);
+    map_t map1(exts1);
+    acc_t acc1(&flag1);
+    stdex::mdspan<int, stdex::dextents<size_t,2>,
+                  Foo::layout_foo, acc_t> m1(Foo::foo_ptr<int>{data1}, map1, acc1);
+    stdex::dextents<size_t, 2> exts2(4,3);
+    map_t map2(exts2);
+    acc_t acc2(&flag2);
+    stdex::mdspan<int, stdex::dextents<size_t,2>,
+                  Foo::layout_foo, acc_t> m2(Foo::foo_ptr<int>{data2}, map2, acc2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.data_handle().data, data1);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.rank(), 2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.rank_dynamic(), 2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.extent(0), 3);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.extent(1), 4);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.static_extent(0), dyn);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.static_extent(1), dyn);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.stride(0), 4);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.stride(1), 1);
+    __MDSPAN_DEVICE_ASSERT_EQ(m1.accessor().flag[0], 9);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.data_handle().data, data2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.rank(), 2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.rank_dynamic(), 2);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.extent(0), 4);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.extent(1), 3);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.static_extent(0), dyn);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.static_extent(1), dyn);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.stride(0), 3);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.stride(1), 1);
+    __MDSPAN_DEVICE_ASSERT_EQ(m2.accessor().flag[0], 7);
```

Comment:
Any good reason to not use the various `operator==` instead?
Do comparisons on the extent, mapping, and accessor instead of all that block that no-one wants to proofread.

Reply:
i guess, technically this is testing mdspan interfaces not whether m2.mapping returns the right thing.